packages/core/src/ctype/CType.chain.ts

```diff
@@ -33,12 +44,107 @@ export function idToChain(cTypeId: ICType['$id']): CTypeHash {
   return idToHash(cTypeId)
 }
 
+// Transform a blockchain-formatted CType input (represented as Bytes) into the original [[IPublicCredentialInput]].
+// It throws if what was written on the chain was garbage.
+function cTypeInputFromChain(input: Bytes): ICType {
+  try {
+    // Throws on invalid JSON input. CType is expected to be a valid JSON document.
+    const reconstructedObject = JSON.parse(input.toUtf8())
+    // Re-compute the ID to validate the resulting ICType.
+    const reconstructedCTypeId = hashToId(getHashForSchema(reconstructedObject))
+    const reconstructedCType: ICType = {
+      ...reconstructedObject,
+      $id: reconstructedCTypeId,
+    }
+    // If throws if the input was a valid JSON but not a valid CType.
+    verifyDataStructure(reconstructedCType)
+    return reconstructedCType
+  } catch {
+    throw new SDKErrors.CTypeError(
+      `The provided payload cannot be parsed as a CType: ${input.toUtf8()}`
+    )
```

Comment:
I believe we should not make this mistake anymore
```suggestion
  } catch (cause) {
    throw new SDKErrors.CTypeError(
      `The provided payload cannot be parsed as a CType: ${input.toUtf8()}`,
      { cause },
    )
```

Reply:
Didn't know about that. https://github.com/KILTprotocol/sdk-js/pull/694/commits/5f48a393298d0bfe172ce96c574c7857b73fe4db